test/unit/Cardano/Wallet/BinarySpec.hs

```diff
@@ -0,0 +1,137 @@
+                ]
+            , outputs =
+                [ TxOut
+                    { address = addr58 "DdzFFzCqrhsug8jKBMV5Cr94hKY4DrbJtkUpqptoGEkovR2QSkcAcRgjnUyegE689qBX6b2kyxyNvCL6mfqiarzRB9TRq8zwJphR31pr"
+                    , coin = Coin 285000000
+                    }
+                , TxOut
+                    { address = addr58 "DdzFFzCqrhsmxmuQpgjUrvRwF5ZKnyQ7pGrS4q53u5B516wcc26maHz9M4myYAkQVc5m9E4DKJjRDjPxuDdK3ZsHb1Dnqf3XorZ1PnzX"
+                    , coin = Coin 1810771919
+                    }
+                ]
+            }
+        ]
+    }
+
+-- A testnet block with a transaction
+block3 :: Block
+block3 = Block
+    { header = BlockHeader
+        { epochIndex = 30
+        , slotNumber = 9278
+        , prevBlockHash = hash16 "b065b5fe97bec5fd130e7a639189499c9d0b1fcf9348c5c19f7a22700da7a35e"
+        }
+    , transactions = Set.fromList
+        [ Tx
+            { inputs =
+                [ TxIn
+                    { txId = hash16 "6967e2b5c3ad5ae07a9bd8d888f1836195a04f7a1cb4b6d083261870068fab1b"
+                    , txIx = 1}
+                , TxIn
+                    { txId = hash16 "7064addc0968bccd7d57d2e7aa1e9c2f666d8387042483fc1e87200cfb96c8f1"
+                    , txIx = 0} ]
+            , outputs =
+                [ TxOut
+                    { address = addr58 "37btjrVyb4KBsw2f3V76ntfwqDPgyf3QmmdsrTSmCnuTGYtS9JgVXzxeQEsKjgWurKoyw9BDNEtLxWtU9znK49SC8bLTirk6YqcAESFxXJkSyXhQKL"
+                    , coin = Coin 1404176490
+                    }
+                , TxOut
+                    { address = addr58 "37btjrVyb4KD5Ne4yvGAHGbQuHUYQX1VPsXh85rBh3UrGSMWdRSFxBYQ9RQRHCMezN6AMLd3uYTC5hbeVTUiPzfQUTCEogg2HrSJKQUjAgsoYZHwT3"
+                    , coin = Coin 1004099328
+                    }
+                ]
+            }
+        ]
+    }
+
+-- | Make a Hash from a Base16 encoded string, without error handling.
+hash16 :: ByteString -> Hash a
+hash16 = either bomb Hash . convertFromBase Base16
+    where
+        bomb msg = error ("Could not decode test string: " <> msg)
+
+-- | Make an Address from a Base58 encoded string, without error handling.
+addr58 :: ByteString -> Address
+addr58 = maybe (error "addr58: Could not decode") Address . decodeBase58 bitcoinAlphabet
+
+spec :: Spec
+spec = do
+    describe "Decoding blocks" $ do
+        it "should decode a block header" $ do
+            bs <- L8.readFile "test/data/block-header-1"
+            let decoded = unsafeDeserialiseFromBytes decodeBlockHeader bs
+            decoded `shouldBe` blockHeader1
+
+        it "should decode a block without transactions" $ do
+            bs <- L8.readFile "test/data/block-1"
+            let decoded = unsafeDeserialiseFromBytes decodeBlock bs
+            decoded `shouldBe` block1
+
+        it "should decode a block with transactions" $ do
+            bs <- L8.readFile "test/data/block-2"
+            let decoded = unsafeDeserialiseFromBytes decodeBlock bs
+            decoded `shouldBe` block2
```

Comment:
Perhaps we should add a test for a block with more than one transaction?

Reply:
Yes we should. We'll find one in cardano explorer and do it in another PR.